### PR TITLE
feat: Setup Quality Gates

### PR DESCRIPTION
**Pull request by manvicha**
Enabling the sbt coverage to evaluate the flaws
in our integration tests. Also, the scalastyle
has been added in order to present a better
standardization format for the code.

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -23,7 +23,10 @@ lazy val commonSettings = Seq(
   },
   compile in Compile := (compile in Compile).dependsOn(checkFormat).value,
   test in Test := (test in Test).dependsOn(checkFormat).value,
-  testOptions in Test += Tests.Argument(TestFrameworks.ScalaTest, "-h", "target/test-reports")
+  testOptions in Test += Tests.Argument(TestFrameworks.ScalaTest, "-h", "target/test-reports"),
+  coverageEnabled := true,
+  coverageMinimum := 0,
```
Comment (manvicha) [inline build.sbt]: I forced the coverageMinimum to 0 in order to pass the coverage and CI

**Review by OstapStrashevskii — APPROVED** [review]
significant!

File: scalastyle-config.xml [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+ <check level="warning" class="org.scalastyle.scalariform.SimplifyBooleanExpressionChecker" enabled="true"></check>
+ <check level="warning" class="org.scalastyle.scalariform.IfBraceChecker" enabled="true">
+  <parameters>
+   <parameter name="singleLineAllowed"><![CDATA[true]]></parameter>
+   <parameter name="doubleLineAllowed"><![CDATA[false]]></parameter>
+  </parameters>
+ </check>
+ <check level="warning" class="org.scalastyle.scalariform.MethodLengthChecker" enabled="true">
+  <parameters>
+   <parameter name="maxLength"><![CDATA[50]]></parameter>
```
Comment (iliaonishchenko) [inline scalastyle-config.xml]: I suppose 50 lines might be not enough. I'd increase it until 100.

File: scalastyle-config.xml [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+ <check level="warning" class="org.scalastyle.scalariform.ObjectNamesChecker" enabled="true">
+  <parameters>
+   <parameter name="regex"><![CDATA[[A-Z][A-Za-z]*]]></parameter>
+  </parameters>
+ </check>
+ <check level="warning" class="org.scalastyle.scalariform.PackageObjectNamesChecker" enabled="true">
+  <parameters>
+   <parameter name="regex"><![CDATA[^[a-z][A-Za-z]*$]]></parameter>
+  </parameters>
+ </check>
```
Comment (iliaonishchenko) [inline scalastyle-config.xml]: I'm not sure what exactly these are checking and why do we need them.

File: project/plugins.sbt [hunk 4]
```diff
@@ -1,3 +1,5 @@
 addSbtPlugin("org.scalameta" % "sbt-scalafmt" % "2.0.6")
 addSbtPlugin("com.typesafe.sbt" % "sbt-native-packager" % "1.6.1")
 addSbtPlugin("com.github.cb372" % "sbt-explicit-dependencies" % "0.2.12")
+addSbtPlugin("org.scoverage" % "sbt-scoverage" % "1.6.1")
```
Comment (iliaonishchenko) [inline project/plugins.sbt:4]: In current configuration we're not able to see results of coverage, unfortunately. 
My proposition is to use this one: https://coveralls.io (via https://github.com/scoverage/sbt-coveralls)
@manvicha you will need to sync with @mnesvit for this. 
@mnesvit could you create an account for us at coveralls.io?

**Review by iliaonishchenko — CHANGES_REQUESTED** [review]
Thank you for you work,
I have several proposition what we can improve in this MR:

File: scalastyle-config.xml (hunk 2 above)
Comment (manvicha) [reply scalastyle-config.xml]: Changed it

File: scalastyle-config.xml (hunk 3 above)
Comment (manvicha) [reply scalastyle-config.xml]: Removed them

File: project/plugins.sbt (hunk 4 above)
Comment (manvicha) [reply project/plugins.sbt:4]: The plugin was added, see this example https://coveralls.io/builds/32961492

**Review by iliaonishchenko — APPROVED** [review]

**Review by mnesvit — APPROVED** [review]

**Pull request merged**